[CI]  SearchPhaseControllerTests » testPartialMergeFailure failure

**Build scan**: https://gradle-enterprise.elastic.co/s/6xnsk44z2zn3a

**Repro line**: `./gradlew ':server:test' --tests "org.elasticsearch.action.search.SearchPhaseControllerTests.testPartialMergeFailure" -Dtests.seed=F035DB3E828C94EB -Dtests.security.manager=true -Dtests.locale=sl-SI -Dtests.timezone=Canada/Central -Druntime.java=11`

**Reproduces locally?**: No

**Applicable branches**: master

**Failure excerpt**: 

```
java.lang.AssertionError: (No message provided) 
at __randomizedtesting.SeedInfo.seed([F035DB3E828C94EB:105FD71CA3DB0A7D]:0) 
at org.junit.Assert.fail(Assert.java:86) 
at org.junit.Assert.assertTrue(Assert.java:41) 
at org.junit.Assert.assertTrue(Assert.java:52)
at org.elasticsearch.action.search.SearchPhaseControllerTests.testPartialMergeFailure(SearchPhaseControllerTests.java:970)
```

--- a/server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java
+++ b/server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java
@@ -278,6 +278,7 @@ class QueryPhaseResultConsumer extends ArraySearchPhaseResults<SearchPhaseResult
                 failure.compareAndSet(null, exc);
                 MergeTask task = runningTask.get();
                 runningTask.compareAndSet(task, null);
+                onPartialMergeFailure.accept(exc);
                 List<MergeTask> toCancel = new ArrayList<>();
                 if (task != null) {
                     toCancel.add(task);
@@ -287,7 +288,6 @@ class QueryPhaseResultConsumer extends ArraySearchPhaseResults<SearchPhaseResult
                 mergeResult = null;
                 toCancel.stream().forEach(MergeTask::cancel);
             }
-            onPartialMergeFailure.accept(exc);
         }
 
         private void onAfterMerge(MergeTask task, MergeResult newResult) {

